tests/test_migrate.py

```diff
@@ -86,3 +97,56 @@ def test_migrate_error_duplicate(
     # no records have been created in the test group
     records = migrator.gn.get_records(query={"facet.q": f"groupOwner/{group_fixture}"})
     assert len(records) == 0
+
+
+def test_migrate_transform_job_id(migrator: Migrator):
+    migrate_batch = migrator.migrate(TransformBatch(), transform_job_id="xxx")
+    assert migrate_batch.transform_job_id == "xxx"
+
+
+def test_migrate_batch_records_success(migrator: Migrator, md_fixtures: list[Fixture]):
+    batch, _ = get_transform_results("noop", migrator)
+    migrate_batch = migrator.migrate(batch, overwrite=False, group=None)
+    for record in migrate_batch.successes():
+        assert record.source_uuid in [f.uuid for f in md_fixtures]
+        assert record.target_uuid not in [f.uuid for f in md_fixtures]
+        assert record.url == GN_TEST_URL
+        assert record.source_content is not None  # TODO: check content when formatting is done
+        assert record.target_content is not None  # TODO: check content when formatting is done
+        assert record.template is False
+
+
+def test_migrate_batch_records_failure(migrator: Migrator, md_fixtures: list[Fixture]):
+    batch, _ = get_transform_results("noop", migrator)
+    with patch("ecospheres_migrator.geonetwork.GeonetworkClient.update_record") as mocked_method:
+        mocked_method.side_effect = Exception("Mocked update_record error")
+        migrate_batch = migrator.migrate(batch, overwrite=False, group=None)
+    for record in migrate_batch.failures():
+        assert record.source_uuid in [f.uuid for f in md_fixtures]
+        assert record.url == GN_TEST_URL
+        assert record.source_content is not None  # TODO: check content when formatting is done
+        assert record.target_content is not None  # TODO: check content when formatting is done
```

Comment:
Should we have target_content on error?

Reply:
The question is rather should we _not_ have it on error ;-)

While not strictly necessary, it might make it easier for further reporting / debugging, like see the payload the caused a problem in migrate. I'd leave it since it's already there.

Reply:
What is currently in `target_content` on errors? We should have the payload that caused problem in `source_content`, is that what you'd also put in `target_content`?

Reply:
`source_content` and `target_content` are copied from transform's `original` and `result` respectively.

Reply:
finally makes sense, somehow :)